File: scripts/codex_webpush_notify.js
#!/usr/bin/env node

const backendUrl = process.env.CODEX_NOTIFY_BACKEND_URL;
const token = process.env.CODEX_NOTIFY_TOKEN;

if (!backendUrl || !token) {
  console.error('CODEX_NOTIFY_BACKEND_URL and CODEX_NOTIFY_TOKEN are required.');
  process.exit(1);
}

const raw = process.argv[2];
if (!raw) {
  process.exit(0);
}

const event = JSON.parse(raw);
if (event.event !== 'agent-turn-complete') {
  process.exit(0);
}

const payload = {
  event: event.event,
  title: event.title || 'Codex task finished',
  summary: event.summary || '',
  status: event.status || 'completed',
  thread_id: event.thread_id,
  cwd: event.cwd,
  client: event.client,
  raw: event
};

const response = await fetch(`${backendUrl}/api/codex/event`, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    Authorization: `Bearer ${token}`
  },
  body: JSON.stringify(payload)
});

if (!response.ok) {
  const text = await response.text();
  console.error('Failed to send Codex event', response.status, text);
  process.exit(1);
}

console.log('Codex event delivered', response.status);

File: scripts/codex_webpush_notify.py
#!/usr/bin/env python3
import json
import os
import sys
import urllib.request

BACKEND_URL = os.environ["CODEX_NOTIFY_BACKEND_URL"]
TOKEN = os.environ["CODEX_NOTIFY_TOKEN"]

def main() -> int:
    event = json.loads(sys.argv[1])
    if event.get("event") != "agent-turn-complete":
        return 0

    payload = {
        "event": event.get("event"),
        "title": event.get("title") or "Codex task finished",
        "summary": event.get("summary") or "",
        "status": event.get("status") or "completed",
        "thread_id": event.get("thread_id"),
        "cwd": event.get("cwd"),
        "client": event.get("client"),
        "finished_at": event.get("finished_at"),
        "raw": event,
    }

    req = urllib.request.Request(
        BACKEND_URL + "/api/codex/event",
        data=json.dumps(payload).encode("utf-8"),
        headers={
            "Content-Type": "application/json",
            "Authorization": f"Bearer {TOKEN}",
        },
        method="POST",
    )

    with urllib.request.urlopen(req) as response:
        print(response.status)

    return 0

if __name__ == "__main__":
    raise SystemExit(main())
